File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit=0, n=prices.size(), maxprofit=0;
        int buy=prices[0];
        for(int i=1;i<n;i++){
            if(buy>prices[i]){
                buy=prices[i];
            }
            profit=prices[i]-buy;
            if(maxprofit<profit){
                maxprofit=profit;
            }
        }
        return maxprofit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n=prices.length;
        int maxprofit=Integer.MIN_VALUE;
        int profit=0;
        int buy=prices[0];
        for(int i=0;i<n;i++){
            buy=Math.min(buy,prices[i]);
            profit=prices[i]-buy;
            if(profit>maxprofit){
                maxprofit=profit;
            }
        }
        return maxprofit;

    }
}